### PR TITLE
Advance search search page

### PR DESCRIPTION
**Pull request by GauravD2t**
## References
* Fixes DSpace/DSpace#8431 (if this fixes an issue ticket)

## Description
The operator drop-down list should have four values: Contains, Not Contains, Equals, and Not Equals.
The operator drop-down list should move above the text field in the Advanced Search section.
The advanced search options should be removed from individual facets (Author, Subject, etc)
## Instructions for Reviewers
Please add a more detailed description of the changes made by your PR. At a minimum, providing a bulleted list of changes in your PR is helpful to reviewers.

List of changes in this PR:
* First, ...
* Second, ...

**Include guidance for how to test or review your PR.** This may include: steps to reproduce a bug, screenshots or description of a new feature, or reasons behind specific changes. 

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [ESLint](https://eslint.org/) validation using `yarn lint`
- [x] My PR doesn't introduce circular dependencies (verified via `yarn check-circ-deps`)
- [x] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [ ] If my PR includes new libraries/dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [ ] If my PR includes new features or configurations, I've provided basic technical documentation in the PR itself.
- [x] If my PR fixes an issue ticket, I've [linked them together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

### CODE REVIEW

File: src/app/shared/search/advanced-search/advanced-search.component.html
Comment (tdonohue) [inline src/app/shared/search/advanced-search/advanced-search.component.html:37]: As [we mentioned previously](https://github.com/DSpace/DSpace/issues/8431#issuecomment-1493684216), the developer team requested that the list of operators be moved **before** the search text field. 

So, this select needs to be moved above the search text field

File: src/app/shared/search/advanced-search/advanced-search.component.html
Comment (tdonohue) [inline src/app/shared/search/advanced-search/advanced-search.component.html:39]: This key `search.filters.operator.equals.text` is missing from our `en.json5` file. It needs to be added.

File: src/assets/i18n/en.json5
Comment (tdonohue) [inline src/assets/i18n/en.json5:5393]: Please don't remove existing keys from this file.  You've accidentally deleted about 16 different keys from the `en.json5` file.  We cannot merge this PR until they are restored.  You should only add **new keys** and not remove ones that exist (the only exception is if you are replacing a key)

File: src/assets/i18n/en.json5
Comment (tdonohue) [inline src/assets/i18n/en.json5:5419]: It appears you are missing `search.fitlers.filter.subject.text` in this file.  It needs to be added.

File: src/assets/i18n/en.json5
Comment (tdonohue) [inline src/assets/i18n/en.json5:5437]: Please move this filter setting up alongside the others

File: src/assets/i18n/en.json5
Comment (tdonohue) [inline src/assets/i18n/en.json5:5447]: Please move this filter setting up alongside all the others.  We want to keep them all grouped together to make them easy to find.

File: src/assets/i18n/en.json5
Comment (tdonohue) [inline src/assets/i18n/en.json5:5435]: This i18n key doesn't seem to be used in this PR?  I think it can be removed maybe?

File: src/assets/i18n/en.json5
Comment (tdonohue) [inline src/assets/i18n/en.json5:5433]: Again, this one doesn't seem to be used in this PR

File: src/app/shared/search/search-filters/search-filters.component.spec.ts
Comment (tdonohue) [inline src/app/shared/search/search-filters/search-filters.component.spec.ts:64]: This shouldn't be commented out.  We need to keep tests uncommented so that they always run.  Please uncomment or explain why it's commented it.

**Review by tdonohue — CHANGES_REQUESTED** [review]
@GauravD2t : I gave this another test and review today.  The feature works overall, but the code still needs some cleanup & fixes.  You need to also make sure that you are not accidentally removing code added by other PRs (see the en.json5 changes in this PR...where you are accidentally removing other code).

Finally, I'm noticing a small bug in how the "Search text" field is doing validation.   Sometimes it shows an error on the "Search text" field immediately after a search is done.  Here's what it looks like (notice the read error highlight): 
error-in-search-text

Here's how to reproduce it:
* In Advance Search, select Author
* Enter some text
* Select Equals or Not Equals
* Click "Add"
* The advanced filter will be added successfully, but the "Search text" field will show an error highlight.  This error highlight should not appear.
* (I should note that I cannot always get this error to appear. It seems random, but it should not be appearing obviously.)

More notes inline below...

File: src/app/shared/search/search-filters/search-filters.component.spec.ts
Comment (GauravD2t) [reply src/app/shared/search/search-filters/search-filters.component.spec.ts:64]: remove commented

File: src/assets/i18n/en.json5
Comment (tdonohue) [inline src/assets/i18n/en.json5:5617]: @GauravD2t : It appears you are still accidently removing i18n keys from `en.json5` which were added by other PRs.  These keys MUST be restored on lines 5605 to 5618 here.

File: src/assets/i18n/en.json5
Comment (tdonohue) [inline src/assets/i18n/en.json5:5591]: Again, here you've accidently removed 3 more i18n keys on line 5587-5591. They must be restored

File: src/assets/i18n/en.json5
Comment (tdonohue) [inline src/assets/i18n/en.json5:5647]: You've again accidently removed another 9 i18n keys here on lines 5631 to 5647.  These must **all** be restored as removing them will break other features.

**Review by tdonohue — CHANGES_REQUESTED** [review]
@GauravD2t : I retested this today and it's working much better now.  I've only found one bug.

* If the Advanced Search returns **no results**, all of the additional "Filters" disappear.  See this screenshot where the "Author", "Subject", Date", etc have all disappeared from the sidebar
filters-gone
    * Here's how I reproduced that
    * Go to the Search page, and perform a "Title" "Equals" search for a title that doesn't exist
    * You'll get no results and all the other filters will disappear from the page.

Beyond that, I also have noticed that there are still i18n keys being **removed** from the `en.json5` file.  Removing those i18n keys causes other features to break. So this MUST be fixed before this PR can be merged.

**Comment by GauravD2t** [comment]
> @GauravD2t : I retested this today and it's working much better now. I've only found one bug.
> 
> * If the Advanced Search returns **no results**, all of the additional "Filters" disappear.  See this screenshot where the "Author", "Subject", Date", etc have all disappeared from the "Filters" sidebar
>   filters-gone
>   
>   * Here's how I reproduced that
>   * Go to the Search page, and perform a "Title" "Equals" search for a title that doesn't exist
>   * You'll get no results and all the other filters will disappear from the page.
> 
> Beyond that, I also have noticed that there are still i18n keys being **removed** from the `en.json5` file. Removing those i18n keys causes other features to break. So this MUST be fixed before this PR can be merged.

filters come base on searching value

**Review by tdonohue — APPROVED** [review]
👍 @GauravD2t : Thanks for the updates & for explaining why the filters disappear when a search fails.  This looks good to me now.  That said, I did need to correct a minor typo in `config.example.yml` as it said `filters` instead of `filter`.  Once this completes it's rebuild, I'll merge it.

**Pull request merged**
